### ISSUE TITLE
Malformed string when decoding from JSON

### ISSUE THREAD

**Issue by vladiulianbogdan** [apple/swift-protobuf#743]
Hi,

I have to create SwiftProtobuf messages from JSON. The JSONs are not created from SwiftProtobuf messages (by using `jsonString()` or `jsonUTF8Data()`), instead they are created using JSONSerialization from Foundation.framework.

The problem appears when the JSON has a base64 value which contains a "/" character. JSONSerialization will escape that character with a "\\". When I try to populate a SwiftProtobuf message with the content from JSON, the JSONScanner will report a `JSONDecodingError.malformedString` error because of the unexpected "\\" character.

Example of JSON incorrectly marked as malformed:

```
{"data":"omdtZXNzYWdlZnBhY2tldGdwYXlsb2FkomVpbmRleBg\/ZGRhdGF4gDAwMDEwMjAzMDQwNTA2MDcwODA5MGEwYjBjMGQwZTBmMTAxMTEyMTMxNDE1MTYxNzE4MTkxYTFiMWMxZDFlMWYyMDIxMjIyMzI0MjUyNjI3MjgyOTJhMmIyYzJkMmUyZjMwMzEzMjMzMzQzNTM2MzczODM5M2EzYjNjM2QzZTNm", "id": 0}
```

I had to fix this in one of my forks in order to unblock my project. In parseBytes method from JSONScanner I am also handling the "\" character.

* what OS you are developing on
macOS 10.13.3

* for macOS, what version of Xcode you are using
Xcode 9.2

* what version of Swift is your code set to compile with
Swift 4.0

* what branch/tag of SwiftProtobuf you are using
1.0.3

**Comment by tbkka** [comment]
Thanks for the report!  Yes, we should accept arbitrary backslash escapes here.  That was definitely an oversight.

**Comment by tbkka** [comment]
Please look at #744 and let me know what you think.

**Comment by tbkka** [comment]
I think this is fixed with PR #744.  Let us know if you still have problems.

**Comment by aarashgoodarzi** [comment]
the problem still exists!!

**Comment by tbkka** [comment]
Can you please open a new Issue with more details?  In particular, I think it might help if you could show us the invalid JSON you're seeing.
